### PR TITLE
Magic Design Focus Proposal

### PR DESCRIPTION
**Pull request by Alastors**
Through this proposal, I would like to attempt to formalize some generalized concepts within Magic design, that being, ensuring future magic content is created with the intentions of the mods respected.

One example of this being, a formal ban on adding numbers to Botania as it is against Vazkii's intentions with the mod, i.e.  if Botania Energistics is ever finished, the terminal wouldn't be allowed to display the exact int of mana the player has.

I push to formalize a not to all future devs who may want to develop for Thaumcraft, "respect the lore when designing, thaumcraft, above all else, is a lore mod above a magic mod. The mod lives and dies on the lore, as such, I would like to formalize this long held informality.

Likewise, I push to formalize some new codes to prevent some mistakes I personally have made in the past, within the design of magic, two examples being "crass usage of swears, purposefully misleading information, and tabs that are just one massive shitpost" (obviously I use more formalized language in the actual legislation.)

### CODE REVIEW

**Review by Connor-Colenso — CHANGES_REQUESTED** [review]
I strongly disagree with adhering strictly to Vazkii's vision, especially within the GTNH context. Given that this pack primarily focuses on tech and engineering, we'll need quantifiable metrics if we integrate more with Botania. 

It would endlessly frustrate players if their AE simply labeled mana quantity as 'a lot'.

**Comment by Alastors** [comment]
> I strongly disagree with adhering strictly to Vazkii's vision, especially within the GTNH context. Given that this pack primarily focuses on tech and engineering, we'll need quantifiable metrics if we integrate more with Botania.
> 
> It would endlessly frustrate players if their AE simply labeled mana quantity as 'a lot'.

Personally I disagree, but I removed that for decision at a later date if we ever actually get to integration.

File: developer's code of conduct.md [hunk 1]
```diff
@@ -16,10 +16,25 @@ To improve review quality, in PR about new feature/feature change/balance change
...
+   2) Thaumcraft
+      1) Thaumcraft is a lore mod, while yes, it is a magic mod, above all else it is trying to tell a distinct story, with both limitations and specific paths. As such, when designing and balancing for Thaumcraft, one must take the lore into consideration, one example being: "Essentia types cannot mix otherwise an explosion would occur." Meaning in no circumstance are essentias allowed to touch to maintain canon. (i.e. a single jar can't store multiple essentia but a reservoir can.) However, a designer could take advantage of this and create an essentia explosive, for example writing lore of a mechanism that works similar to a bombardier beetle.
+      2) Thaumcraft lives and dies on the quality of the lore, while there are some allowances for data here and there, hard numbers are meant for the questbook, not the thaumonomicon. 
+      3) Create lore entries with the upmost seriousness, joke entries ***are*** allowed and are encouraged to an extent, however, this is meant for singular isolated items and chains, please do not create an entire research tab for trivial things or jokes.
```
Comment (Pxx500) [inline developer's code of conduct.md:28]: maybe delete the part that says "encouraged", if they are sometimes allowed there is no need for them to be encouraged

Comment (Alastors) [reply developer's code of conduct.md]: done

**Comment by bombcar** [comment]
> I strongly disagree with adhering strictly to Vazkii's vision, especially within the GTNH context. Given that this pack primarily focuses on tech and engineering, we'll need quantifiable metrics if we integrate more with Botania.
> 
> It would endlessly frustrate players if their AE simply labeled mana quantity as 'a lot'.

I think there's a line somewhere, where (and this is all hand wavy) "GT machines" interacting with magic have GT-style interfaces and numbers, where as the actual "magic" (like botania) maintain their consistency. 

Of course, you run the risk of making the "correct magic way" of doing things entirely more complicated/boring/annoying than tech, and so tech wins out, so it should be very sparing.

But the example, once mana is loaded in to AE2 somehow, it's in the realm of tech.

But that's a discussion for another time; what is currently here looks good to me.

**Comment by glowredman** [comment]
> I strongly disagree with adhering strictly to Vazkii's vision, especially within the GTNH context. Given that this pack primarily focuses on tech and engineering, we'll need quantifiable metrics if we integrate more with Botania. 

I don't think the two points contradict. We can totally balance based on actual number while purposefully hiding them from the player. Of course there will be those, who look in the code but that has always been possible.

> It would endlessly frustrate players if their AE simply labeled mana quantity as 'a lot'.

I think that depends on how it's done. If there is just a String telling you "full"/"mostly full"/etc. that is bad design, but e.g. using bars like already used in Botania, I see less of a problem. Yes, there will always be people who just have to have numbers but maybe Botania just isn't the mod for them then.

**Comment by bombcar** [comment]
> Yes, there will always be people who just have to have numbers but maybe Botania just isn't the mod for them then.

Which is the key, going back to the "original problem statement" and working out what the player is trying to do, and solving that problem in the context of the magic mod's design. *Sometimes* you may have to fall back to something outside the design, but it should be carefully considered.

**Review by bombcar — APPROVED** [review]
Looks good to me, though 8/2/1/6 could be a more general rule and not just thaum.

**Pull request merged**